### ISSUE TITLE
System.Object incompatibility between csx files, .NETCore, and Microsoft.CodeAnalysis v2.0.0-beta1

### ISSUE THREAD

**Issue by ljw1004** [dotnet/roslyn#12419]
1. File > New > C# > .NETCore > ConsoleApp "ConsoleApp1"
2. File > New > C# > .NETCore > ConsoleApp "TestApp"
3. Inside TestApp\project.json, change the imports `"netcoreapp1.0": { "imports": "portable-net45+win8" }`
4. Add two NuGet references `"Microsoft.CodeAnalysis": "2.0.0-beta1"`, `"Microsoft.DotNet.ProjectModel.Workspaces": "1.0.0-preview2-003121"`
5. Inside TestApp, add this code, and run it:

```csharp
var workspace = new ProjectJsonWorkspace(@"<path>\ConsoleApp1");
var solution = workspace.CurrentSolution; 
var project = solution.Projects.Single();
project = project.AddDocument("a.csx", "").WithSourceCodeKind(SourceCodeKind.Script).Project;
var comp = project.GetCompilationAsync().Result;
foreach (var d in comp.GetDiagnostics().Where(d => d.Severity == DiagnosticSeverity.Error))
   Console.WriteLine(d.GetMessage());
```

**Expected Behavior**:

It should run without printing any diagnostics

**Actual Behavior**:

The `comp.GetDiagnostics()` call retrieved this diagnostic:
```
The type or namespace name 'System.Object, System.Private.CoreLib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=7cec85d7bea7798e' could not be found in the global namespace (are you missing an assembly reference?)
```

**Notes**:

I get the expected behavior if I make *any* of the following changes...

1. Paste+run the above code from a desktop console app (rather than a .NETCore console app)
2. Run it from a .NETCore console app but change it to `.WithSourceCodeKind(SourceCodeKind.Regular)`
3. Delete the line which adds a document
4. Downgrade my reference to Microsoft.CodeAnalysis `v1.3.2`

**Comment by ljw1004** [comment]
Maybe related to #12393 ?

**Comment by tmat** [comment]
Could you share ```project.json``` for both apps?

**Comment by ljw1004** [comment]
Here are the two projects zipped up: [repro.zip](https://github.com/dotnet/roslyn/files/356380/repro.zip)

TestApp\project.json:
```json
{
  "version": "1.0.0-*",
  "buildOptions": {
    "emitEntryPoint": true
  },

  "dependencies": {
    "Microsoft.CodeAnalysis": "2.0.0-beta1",
    "Microsoft.DotNet.ProjectModel.Workspaces": "1.0.0-preview2-003121",
    "Microsoft.NETCore.App": {
      "type": "platform",
      "version": "1.0.0"
    }
  },

  "frameworks": {
    "netcoreapp1.0": {
      "imports": "portable-net45+win8"
    }
  }
}
```

ConsoleApp1\project.json:
```json
{
  "version": "1.0.0-*",
  "buildOptions": {
    "emitEntryPoint": true
  },

  "dependencies": {
    "Microsoft.NETCore.App": {
      "type": "platform",
      "version": "1.0.0"
    }
  },

  "frameworks": {
    "netcoreapp1.0": {
      "imports": "dnxcore50"
    }
  }
}
```

**Comment by tmat** [comment]
Odd, I'm getting error

```
Errors in c:\temp\projects\ConsoleApp2\src\TestApp\TestApp.xproj
    Package Microsoft.DotNet.ProjectModel.Workspaces 1.0.0-preview2-003121 is not compatible with netcoreapp1.0 (.NETCoreApp,Version=v1.0). Package Microsoft.DotNet.ProjectModel.Workspaces 1.0.0-preview2-003121 supports:
      - net451 (.NETFramework,Version=v4.5.1)
      - netstandard1.6 (.NETStandard,Version=v1.6)
    One or more packages are incompatible with .NETCoreApp,Version=v1.0.
```

when restoring packages in VS 2015 U3.

**Comment by ljw1004** [comment]
Sorry, I'd missed a step...

* Inside TestApp\project.json, change the imports `"netcoreapp1.0": { "imports": "portable-net45+win8" }`

**Comment by tmat** [comment]
I believe this issue has been fixed a while ago.
Try using "Microsoft.CodeAnalysis": "2.0.0-beta4-60709-02" from https://dotnet.myget.org/F/roslyn-master-nightly/api/v3/index.json feed.

**Comment by ljw1004** [comment]
Confirmed, it's fixed in `2.0.0-beta4-60711-01`. Thanks Tomas.
